From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: add struct bpf_map back
Patch-mainline: No, SUSE-specific
References: bsc#1098425

[js] removed by c93552c443ebc63b14e26e46d2e76941c88e0d71. Hopefully
     nobody was using map_ptr?

Signed-off-by: Gary Lin <glin@suse.com> 

---
 include/linux/bpf_verifier.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -123,8 +123,12 @@ struct bpf_verifier_state_list {
 struct bpf_insn_aux_data {
 	union {
 		enum bpf_reg_type ptr_type;	/* pointer type for load/store insns */
+#ifndef __GENKSYMS__
 		unsigned long map_state;	/* pointer/poison value for maps */
 		u32 alu_limit;			/* limit for add/sub register with pointer */
+#else
+		struct bpf_map *map_ptr;        /* pointer for call insn into lookup_elem */
+#endif
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
 	bool seen; /* this insn was processed by the verifier */
